DoSomething/gambit-conversations
# Broadcast User validation

#### What's this PR do?

* Adds config variables for toggling Broadcast Settings with v1/v2 data

* Loads and validates User for a `POST /v2/messages?origin=broadcast` request, using the existing middleware used in `POST /v1/send-message`.

* Changes expected body param in v2 Broadcast Messages from `mobile` to `northstarId`. 


* Uses a new `platform` field on our Broadcast content type to determine the Conversation platform for the given v2 Broadcast message

* Adds tests for User Get and Validate middleware, Import Message params middleware

#### How should this be reviewed?

* Confirm the `DS_GAMBIT_CONVERSATIONS_BROADCASTS_USE_V2` config variable toggles the expected data for the `data.webhook` in a `GET /api/v1/broadcasts/:id` request.

   * When `DS_GAMBIT_CONVERSATIONS_BROADCASTS_USE_V2` is set, the URL should be set to `DS_BLINK_GAMBIT_BROADCAST_WEBHOOK_URL`, and return expected body with `northstarId`, `broadcastId`

   * When `DS_GAMBIT_CONVERSATIONS_BROADCASTS_USE_V2` is not set, the URL should be set to `DS_BLINK_SMS_BROADCAST_WEBHOOK_URL`,  and return expected body with `To`, `From`


* Confirm v2 Broadcast works as expected:

    * Make sure your NS user `sms_status` is set to `'active'` or `'less'`

    * [`POST /v2/messages?origin=broadcast` ](https://github.com/DoSomething/gambit-conversations/blob/feature/271-validation/documentation/endpoints/messages.md#broadcast)with your northstarId, and a broadcastId. Confirm message is delivered

    * `POST /v1/receive-message ` using  your mobile number, and sending `Body` with an unsubcribe trigger like `'stop'`

    * `POST /v2/messages?broadcast` with your northstarID, and a broadcastId. Confirm 422 error is returned, because your User is unsubscribed

* Confirm v1 Broadcast works as expected:

    * `POST /v1/import-mesage?broadcastId=:broadcastId` to verify expected behavior (message is imported, but not posted to Twilio. no check for User `sms_status`)

Ideally we won't need to support v1 much longer. It's live as a backup plan and to make migration easier.

#### Any background context you want to provide?
https://github.com/DoSomething/blink/pull/187 makes the corresponding changes to expected `CustomerIoGambitBroadcast` messages.

#### Relevant tickets
* Fixes #271 
* Completes todo for broadcasts in #239 

#### Checklist
